addon-test-support/ember-mocha/setup-test.js

```diff
@@ -12,7 +12,12 @@ import { resolve } from 'rsvp';
 const _assign = assign || merge;
 
 function chainHooks(hooks, context) {
-  return hooks.reduce((promise, fn) => promise.then(fn.bind(context)), resolve());
+  return hooks.reduce((promise, fn) => {
+    return promise
+      .then(fn.bind(context))
+      // eslint-disable-next-line no-console
+      .catch((error) => console.error('Hook throws an error: ', error));
```

Comment:
Hmm, this will prevent getting a "break on exceptions" style breakpoint in your offending hook...

Can you double check?

Reply:
Pause on exception is working even if the exception is caught.

<img width="1005" alt="screen shot 2018-07-27 at 22 58 41" src="https://user-images.githubusercontent.com/1788571/43343913-ac6a8236-91f0-11e8-8724-4dfedaba7fc3.png">

<img width="859" alt="screen shot 2018-07-27 at 23 01 56" src="https://user-images.githubusercontent.com/1788571/43344067-292ddb6a-91f1-11e8-893a-2be212be5ff4.png">

@rwjblue let me know if you have a better idea, how we can do this.

Reply:
In the screenshot you are using the checkbox for “pause on caught exceptions”. Maybe that’s just what you always have had to do (due to mochas usage of exceptions for flow control)?

Reply:
Sorry, it looks like I misunderstood you.

One thing that we can do is to just make sure that the teardown happens in all cases. Something like

```javascript
afterEach(function() {
  return chainHooks(afterEachHooks, this)
    .finally(() => teardownContext(this))
    .finally(() => {
      // delete any extraneous properties
      for (let key in this) {
        if (!(key in originalContext)) {
          delete this[key];
        }
      }

      // copy over the original values
      _assign(this, originalContext);
    });
});
```

This way the errors that happen in the hooks will be returned to mocha, but we would have cleared the test context.

Reply:
<img width="994" alt="screen shot 2018-07-28 at 11 30 31" src="https://user-images.githubusercontent.com/1788571/43354696-add131ce-9259-11e8-88cb-07be4cc15011.png">

Reply:
Not a fan of the idea?